DAO8

Max-Pochepets opened the pull request:
DataBase web description complete

Max-Pochepets committed:
Initial

Max-Pochepets committed:
Fix:

url manipulations

Max-Pochepets committed:
Fix:

*previous commit*
same with drivers and manufacturers

semyonich commented on src/main/webapp/WEB-INF/views/manufacturers/create.jsp:
```diff
@@ -0,0 +1,22 @@
+        <form method="post" action="${pageContext.request.contextPath}/manufacturers/create">
+            Manufacturer's name:<input type="text" name="name">
+            Manufacturer's country:<input type="text" name="country">
+
```
IMHO you should check input data for non empty value

semyonich replied:
check input

semyonich commented on src/main/webapp/WEB-INF/views/drivers/create.jsp:
```diff
@@ -0,0 +1,22 @@
+
+        <form method="post" action="${pageContext.request.contextPath}/drivers/create">
+            Driver's name:<input type="text" name="name">
+            Driver's license number:<input type="text" name="license">
```
same

semyonich replied:
check input

semyonich commented on src/main/webapp/WEB-INF/views/cars/drivers/delete.jsp:
```diff
@@ -0,0 +1,20 @@
+    <form method="post" action="${pageContext.request.contextPath}/cars/drivers/delete">
+        Car's id:<input type="text" name="carID">
+        Driver's id:<input type="text" name="driverID">
+
```
what if carId will be "test" ? check input

semyonich commented on src/main/java/crud/controller/driver/GetAllDriversController.java:
```diff
@@ -0,0 +1,25 @@
+
+public class GetAllDriversController extends HttpServlet {
+    private static final Injector INJECTOR = Injector.getInstance("crud");
+    private static final DriverService driverService
```
it's not a CONSTANT, use common camelCase

Max-Pochepets replied:
maybe I've misunderstood, but why is it not?

semyonich commented on src/main/java/crud/controller/driver/DeleteDriversController.java:
```diff
@@ -0,0 +1,23 @@
+
+public class DeleteDriversController extends HttpServlet {
+    private static final Injector INJECTOR = Injector.getInstance("crud");
+    private static final DriverService driverService
```
same

semyonich commented on src/main/java/crud/controller/driver/CreateDriversController.java:
```diff
@@ -0,0 +1,31 @@
+
+public class CreateDriversController extends HttpServlet {
+    private static final Injector INJECTOR = Injector.getInstance("crud");
+    private static final DriverService driverService
```
check it everywhere

Max-Pochepets committed:
Fix:

input data checks added;
invalid data situations handled;
constant variables named properly;

Max-Pochepets committed:
Refactor: unused imports deleted

Max-Pochepets committed:
Fix: url manipulations

Max-Pochepets committed:
Fix: possibility of empty input handled

dmytro-hryhoruk commented on src/main/java/crud/controller/ClearDataController.java:
```diff
@@ -0,0 +1,16 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ClearDataController extends HttpServlet {
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException {
+        ConnectionUtil.clearTable();
+        resp.sendRedirect(req.getContextPath() + "/");
+    }
```
I don't think that you should have such functionality in your project
Imagine : https://rozetka.com.ua/ua/computers-notebooks/clear

Max-Pochepets replied:
for the purpose of testing

dmytro-hryhoruk commented on src/main/java/crud/controller/InjectDataController.java:
```diff
@@ -0,0 +1,59 @@
+    private static final CarService CAR_SERVICE
+            = (CarService) INJECTOR.getInstance(CarService.class);
+    private static final DriverService DRIVER_SERVICE
+            = (DriverService) INJECTOR.getInstance(DriverService.class);
```
nither of those are constants

dmytro-hryhoruk commented on src/main/java/crud/controller/car/AddDriverToCarController.java:
```diff
@@ -0,0 +1,43 @@
+import javax.servlet.http.HttpServletResponse;
+
+public class AddDriverToCarController extends HttpServlet {
+    private static final Injector INJECTOR = Injector.getInstance("crud");
+    private static final CarService CAR_SERVICE
+            = (CarService) INJECTOR.getInstance(CarService.class);
+    private static final DriverService DRIVER_SERVICE
+            = (DriverService) INJECTOR.getInstance(DriverService.class);
```
same

Max-Pochepets replied:
why?

Max-Pochepets replied:
nvm, I got it)

dmytro-hryhoruk commented on src/main/java/crud/controller/car/CreateCarsController.java:
```diff
@@ -0,0 +1,43 @@
+    private static final Injector INJECTOR = Injector.getInstance("crud");
+    private static final CarService CAR_SERVICE
+            = (CarService) INJECTOR.getInstance(CarService.class);
+    private static final ManufacturerService MANUFACTURER_SERVICE
```
same

dmytro-hryhoruk commented on src/main/java/crud/controller/car/DeleteDriverFromCar.java:
```diff
@@ -0,0 +1,43 @@
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        try {
+            String carID = req.getParameter("carID");
```
```suggestion
            String carID = req.getParameter("car_id");
```
Change all parameters name in such manner

dmytro-hryhoruk commented on src/main/java/crud/controller/driver/CreateDriversController.java:
```diff
@@ -0,0 +1,31 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class CreateDriversController extends HttpServlet {
```
```suggestion
public class CreateDriverController extends HttpServlet {
```

dmytro-hryhoruk commented on src/main/java/crud/controller/driver/CreateDriversController.java:
```diff
@@ -0,0 +1,31 @@
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException {
+        String name = req.getParameter("name");
+        String license = req.getParameter("license");
```
```suggestion
        String name = req.getParameter("driver_name");
        String license = req.getParameter("driver_license");
```

dmytro-hryhoruk commented on src/main/java/crud/controller/driver/DeleteDriversController.java:
```diff
@@ -0,0 +1,23 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class DeleteDriversController extends HttpServlet {
```
```suggestion
public class DeleteDriverController extends HttpServlet {
```

dmytro-hryhoruk commented on src/main/java/crud/controller/manufacturer/DeleteManufacturersController.java:
```diff
@@ -0,0 +1,23 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class DeleteManufacturersController extends HttpServlet {
```
```suggestion
public class DeleteManufacturerController extends HttpServlet {
```

dmytro-hryhoruk commented on src/main/webapp/WEB-INF/views/cars/all.jsp:
```diff
@@ -0,0 +1,69 @@
+                    <td>
+                        <c:out value="${car.manufacturer.country}"/>
+                    </td>
+                    <td>
+                        <c:forEach var="driver" items="${car.drivers}">
+                            <c:out value="${driver.name}"/>
+                            <br />
+                        </c:forEach>
+                    </td>
+                    <td>
+                        <c:forEach var="driver" items="${car.drivers}">
+                            <c:out value="${driver.licenseNumber}"/>
+                            <br />
+                        </c:forEach>
+                    </td>
```
could you do this with one loop?

Max-Pochepets replied:
it will look awful

dmytro-hryhoruk commented on src/main/java/crud/controller/car/DeleteDriverFromCar.java:
```diff
@@ -0,0 +1,43 @@
+            Driver driver = driverService.get(Long.valueOf(driverID));
+            CAR_SERVICE.removeDriverFromCar(driver, car);
+            resp.sendRedirect(req.getContextPath() + "/cars/");
+        } catch (NoSuchElementException e) {
+            req.setAttribute("message", "Invalid data. Please check if car/driver exists.");
+            req.getRequestDispatcher("/WEB-INF/views/cars/drivers/delete.jsp").forward(req, resp);
```
let's not catch runtime exceptions. In the scope of this task we can just believe that we will get the real id's from the frontend

Max-Pochepets committed:
Fix:

naming fixed
services and data to be injected changed from global to local scope
clearing data functionality removed

Max-Pochepets committed:
Fix:

data validation removed
